ext/http/01_http.js

```diff
@@ -269,35 +269,50 @@
           ) {
             throw new TypeError("Unreachable");
           }
-          const reader = respBody.getReader();
-          while (true) {
-            const { value, done } = await reader.read();
-            if (done) break;
-            if (!ObjectPrototypeIsPrototypeOf(Uint8ArrayPrototype, value)) {
-              await reader.cancel(new TypeError("Value not a Uint8Array"));
-              break;
+          const resourceRid = getReadableStreamRid(respBody);
+          if (resourceRid) {
+            if (respBody.locked) {
+              throw new TypeError("ReadableStream is locked.");
             }
+            const _reader = respBody.getReader(); // Aquire JS lock.
+            await core.opAsync(
+              "op_http_write_resource",
+              streamRid,
+              resourceRid,
+            );
+            readableStreamClose(respBody); // Release JS lock.
```

Comment:
@littledivy if the above statement throws should this still be released?

Reply:
oh nice catch, yeah it should call `reader.cancel`

Reply:
Cool. We can fix it when getting this PR back into main.

Reply:
Fixed in https://github.com/denoland/deno/pull/14345